pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,995 @@
+        max_iterations : {None, int}
+            If samples have not previously been written to the file, a new
+            dataset will be created. By default, the size of this dataset will
+            be whatever the length of the sampler's chain is at this point. If
+            you intend to run more iterations, set this value to that size so
+            that the array in the file will be large enough to accomodate
+            future data.
+        """
+        # chain is ntemps x nwalkers x niterations x ndim
+        samples = self.chain
+        ntemps, nwalkers, niterations, _ = samples.shape
+
+        group = fp.samples_group + '/{name}/temp{tk}/walker{wi}'
+
+        # create an empty array if desired, in case this is the first time
+        # writing
+        if max_iterations is not None:
+            if max_iterations < niterations:
+                raise IndexError("The provided max size is less than the "
+                    "number of iterations")
+            out = numpy.zeros(max_iterations, dtype=samples.dtype)
+
+        # create indices for faster sub-looping
+        widx = numpy.arange(nwalkers)
+        tidx = numpy.arange(ntemps)
+
+        # loop over number of dimensions
+        for pi,param in enumerate(self.variable_args):
+            # loop over number of temps
+            for tk in tidx:
+                # loop over number of walkers
+                for wi in widx:
+                    dataset_name = group.format(name=param, tk=tk, wi=wi)
+                    try:
+                        fp[dataset_name][:niterations] = samples[tk,wi,:,pi]
+                    except KeyError:
+                        # dataset doesn't exist yet, see if a larger array is
+                        # desired
+                        if max_iterations is not None:
+                            out[:niterations] = samples[tk,wi,:,pi]
+                            fp[dataset_name] = out
+                        else:
+                            fp[dataset_name] = samples[tk,wi,:,pi]
+
+    def write_likelihood_stats(self, fp, max_iterations=None):
+        """Writes the given likelihood array to the given file. Results are
+        written to: `fp[fp.stats_group/{field}/temp{k}/walker{i}]`, where
+        `{field}` is the name of stat (`loglr`, `prior`), `{k}` is a
+        temperature index (smaller = colder) and `{i}` is the index of a
+        walker.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            A file handler to an open inference file.
+        max_iterations : {None, int}
+            See `write_chain` for details.
+        """
+        lls = self.likelihood_stats
+        ntemps, nwalkers, niterations = lls.shape
+
+        group = fp.stats_group + '/{field}/temp{tk}/walker{wi}'
+
+        if max_iterations is not None and max_iterations < niterations:
+            raise IndexError("The provided max size is less than the "
+                "number of iterations")
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->